source/material_model/visco_plastic.cc

```diff
@@ -1175,6 +1175,24 @@ namespace aspect
                          ExcMessage("If adiabatic heating is enabled you should not add another adiabatic gradient"
                                     "to the temperature for computing the viscosity, because the ambient"
                                     "temperature profile already includes the adiabatic gradient."));
+
+          // Currently, it only makes sense to use this material model with strain weakening when the
+          // nonlinear solver scheme does a single advection iteration. More than one nonlinear advection
+          // iteration will result in the incorrect value of strain being used in the material model, as 
+          // the compositional fields representing strain are updated through the reaction rate terms.
```

Comment:
This should be reaction terms; you want to make clear that the material model uses the reaction terms as opposed to the reaction rate outputs, which are used with operator splitting and would allow it to have more than one advection iteration.